public/models/GameMap.js

```diff
@@ -30,105 +30,117 @@ class GameMap {
-    draw (ctx, player, canvasWidth, canvasHeight) {
-        for(let row = 0; row < this.tiles.length; row++) {
-            for(let column = 0; column < this.tiles[row].length; column++) {
-                switch(this.tiles[row][column]) {
-                    case 0:
-                    //draw path under the player
-                    case 3:
-                        path.draw(ctx, 
-                            (canvasWidth - player.width) / 2 - player.x + (column * this.tileWidth), 
-                            ((canvasHeight - player.height) / 2) - player.y + (row + 1) * this.tileHeight, 
-                            this.tileWidth, this.tileHeight
-                        );
-                        break;
-                    //coin match
-                    case String(this.tiles[row][column]).match(/^4/)?.input:
-                        path.draw(ctx, 
-                            (canvasWidth - player.width) / 2 - player.x + (column * this.tileWidth), 
-                            ((canvasHeight - player.height) / 2) - player.y + (row + 1) * this.tileHeight, 
-                            this.tileWidth, this.tileHeight
-                        );
-                        //take one coin to draw as they are the same
-                        coin.draw(ctx, 
-                            (canvasWidth - player.width) / 2 - player.x + (column * this.tileWidth), 
-                            ((canvasHeight - player.height) / 2) - player.y + (row + 1) * this.tileHeight,
-                            this.tileWidth, this.tileHeight
-                        );
-                        break;
-                    case 1:
-                        wall.draw(ctx, 
-                            (canvasWidth - player.width) / 2 - player.x + (column * this.tileWidth), 
-                            ((canvasHeight - player.height) / 2) - player.y + (row + 1) * this.tileHeight, 
-                            this.tileWidth, this.tileHeight
-                        );
-                        break;
-                    case 2:
-                        goal.draw(ctx, 
-                            (canvasWidth - player.width) / 2 - player.x + (column * this.tileWidth), 
-                            ((canvasHeight - player.height) / 2) - player.y + (row + 1) * this.tileHeight, 
-                            this.tileWidth, this.tileHeight
-                        );
-                        break;
-                    default:
-                }
-            }
-          }
+  // Marianna
+  // draw map in relation to player
+  draw(ctx, player, canvasWidth, canvasHeight) {
+    for (let row = 0; row < this.tiles.length; row++) {
+      for (let column = 0; column < this.tiles[row].length; column++) {
+        switch (this.tiles[row][column]) {
+          case 0:
+            // draw path under the player
+          case 3:
+            path.draw(
+              ctx,
+              (canvasWidth - player.width) / 2 - player.x + (column * this.tileWidth),
```

Comment:
could you explain me this calculations?

Reply:
It's basically same as in the camera video. First of all the player is always draw in the middle which is this first part **(canvasWidth - player.width) / 2**. And then the 2nd part is position of the block in relation to player so firstly the value that we have - x position of player (**- player.x**) so we get the relative position to the player if the x coordinate of our block was 0. But as it is not 0 we need to consider it and that's why this is added **+ (column * this.tileWidth)** which is the current x coordinate of the object. And then the same is done for Y coordinate.